src/debug/dwarf/unit.go

```diff
@@ -62,12 +62,24 @@ func (d *Data) parseUnits() ([]unit, error) {
 		var n Offset
 		n, u.is64 = b.unitLength()
 		vers := b.uint16()
+		// Count the bytes of header read, not counting size
+		var hbcount uint
+		hbcount = 2 // Not counting size
 		if vers != 2 && vers != 3 && vers != 4 {
 			b.error("unsupported DWARF version " + strconv.Itoa(int(vers)))
 			break
 		}
 		u.vers = int(vers)
-		atable, err := d.parseAbbrev(b.uint32(), u.vers)
+		var aoff uint32
+		if !u.is64 {
+			aoff = b.uint32()
+			hbcount += 4
+		} else {
+			// For now, ignore the high bits
+			aoff = uint32(b.uint64())
```

Comment:
I'd recommend checking for overflow here. See the equivalent code in Data.parseTypes.